src/components/ShoppingCart.tsx

```diff
@@ -0,0 +1,31 @@
+import { Offcanvas, Stack } from "react-bootstrap";
+import { useShoppingCart } from "../context/ShoppingCartContext";
+import { CartItem } from "./CartItem";
+import storeItems from "../data/items.json";
+
+type ShoppingCartProps = {
+  isOpen: boolean;
+};
+
+export function ShoppingCart({ isOpen }: ShoppingCartProps) {
+  const { closeCart, cartItems } = useShoppingCart();
+  const total = cartItems.reduce((total, cartItem) => {
+    const item = storeItems.find((i) => i.id === cartItem.id);
+    return total + (item?.price || 0) * cartItem.quantity;
+  }, 0);
```

Comment:
cartTotal can also be a part of the context

Reply:
Resolved